src/main/java/es/tid/pce/pcep/objects/tlvs/SRPolicyCandidatePathIdentifiersTLV.java

```diff
@@ -0,0 +1,196 @@
+		return reserved;
+	}
+
+	public void setReserved(int reserved) {
+		this.reserved = reserved;
+	}
+
+	public long getOriginatorASN() {
+		return originatorASN;
+	}
+
+	public void setOriginatorASN(long originatorASN) {
+		this.originatorASN = originatorASN;
+	}
+
+	public Inet4Address getOriginatorAddress() {
+		return originatorAddress;
+	}
+
+	public void setOriginatorAddress(Inet4Address originatorAddress) {
+		this.originatorAddress = originatorAddress;
+	}
+
+	public long getDiscriminator() {
+		return discriminator;
+	}
+
+	public void setDiscriminator(long discriminator) {
+		this.discriminator = discriminator;
+	}
+	
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = super.hashCode();
+		result = prime * result + Objects.hash(discriminator, originatorASN, originatorAddress, protocol, reserved);
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (!super.equals(obj))
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		SRPolicyCandidatePathIdentifiersTLV other = (SRPolicyCandidatePathIdentifiersTLV) obj;
+		return discriminator == other.discriminator && originatorASN == other.originatorASN
+				&& Objects.equals(originatorAddress, other.originatorAddress) && protocol == other.protocol
+				&& reserved == other.reserved;
+	}
+	
+
+	@Override
+	public String toString() {
+		return "SRPolicyCandidatePathIdentifiersTLV [protocol=" + protocol + ", reserved=" + reserved
+				+ ", originatorASN=" + originatorASN + ", originatorAddress=" + originatorAddress + ", discriminator="
+				+ discriminator + "]";
+	}
+
+	@Override
+	public void encode() {
+		//Log.debug("Encoding SRPolicyCandidatePathIdentifiers TLV");
+		this.setTLVValueLength(16);
+		this.tlv_bytes=new byte[this.getTotalTLVLength()];
+		this.encodeHeader();
+		int offset=4;
+		this.reserved=0;
+		
+		ByteHandler.encode1byteInteger(protocol, this.tlv_bytes, offset);
+		offset+=1;
+		ByteHandler.encode3bytesInteger(reserved, this.tlv_bytes, offset);
+		offset+=3;
+		ByteHandler.encode4bytesLong(originatorASN, this.tlv_bytes, offset);
+		offset+=4;
+		System.arraycopy(originatorAddress.getAddress(),0,this.tlv_bytes,offset,4);
+		offset+=4;
+		ByteHandler.encode4bytesLong(discriminator, this.tlv_bytes, offset);
+
+		
+	}
+	
+
+	public void decode() throws MalformedPCEPObjectException {
+		log.debug("Decoding SRPolicyCandidatePathIdentifiersTLV TLV");
+		int offset=4;//Position of the next subobject
+		if (this.getTLVValueLength()==0)
+		{
+			throw new MalformedPCEPObjectException();
+		}
+			
+		try {
+			this.protocol = ByteHandler.decode1byteInteger(this.tlv_bytes, offset);
+			offset+=1;
+			this.reserved = ByteHandler.decode3bytesInteger(this.tlv_bytes, offset);
+			offset+=3;
+			this.originatorASN = ByteHandler.decode4bytesLong(this.tlv_bytes, offset);
+			offset+=4;
+			byte[] ip=new byte[4]; 
+			System.arraycopy(this.tlv_bytes,offset, ip, 0, 4);
```

Comment:
Be careful when changing to send 128 bits for the originator. By now ignore all the 96 upper bits...